File: storio-sqlite-annotations-processor/src/main/java/com/pushtorefresh/storio/sqlite/annotations/processor/ProcessingException.java
package com.pushtorefresh.storio.sqlite.annotations.processor;

import org.jetbrains.annotations.NotNull;

import javax.lang.model.element.Element;

/**
 * Useful for logging errors from AnnotationProcessor,
 * stores reference to {@link Element} that caused exception so IDE will show developer where is the problem
 */
public class ProcessingException extends RuntimeException {

    /**
     * Element that caused exception
     */
    @NotNull
    private final Element element;

    public ProcessingException(@NotNull Element element, @NotNull String message) {
        super(message);
        this.element = element;
    }

    /**
     * @return not-null element that caused exception
     */
    @NotNull
    public Element element() {
        return element;
    }
}

File: storio-common-annotations-processor/src/main/kotlin/com/pushtorefresh/storio3/common/annotations/processor/ProcessingException.kt
package com.pushtorefresh.storio3.common.annotations.processor

import javax.lang.model.element.Element

/**
 * Useful for logging errors from AnnotationProcessor, stores reference to [Element] that caused
 * exception so IDE will show developer where is the problem
 */
class ProcessingException(val element: Element, message: String) : RuntimeException(message)